scripts/collect_metadata.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"sync"
+
+	"github.com/go-playground/validator/v10"
+	"golang.org/x/mod/semver"
+)
+
+var validate *validator.Validate
+
+type Connector struct {
+	Name        string      `json:"name" validate:"required"`
+	Version     string      `json:"version" validate:"required,semver"`
+	Description string      `json:"description" validate:"required"`
+	Author      string      `json:"author" validate:"required"`
+	Provider    interface{} `json:"provider" validate:"required"`
+}
+
+type ConnectorMetadata struct {
+	Name        string   `json:"name"`
+	Description string   `json:"description"`
+	Author      string   `json:"author"`
+	Latest      string   `json:"latest"`
+	Versions    []string `json:"versions"`
+	Url         string   `json:"url"`
+}
+
+type ConnectorsMetadata map[string]*ConnectorMetadata
+type CollectMetadata struct {
+	rootPath       string
+	metaPath       string
+	storagePath    string
+	connectorPaths []string
+}
+
+func NewCollectMetadata(
+	rootPath string,
+	metadataFilePath string,
+	storageFolderPath string,
+	connectorPaths []string,
+) *CollectMetadata {
+	cp := make([]string, len(connectorPaths))
```

Comment:
I know it's idiomatic, but I prefer long-name variables over single-letter or acronyms as variable names. In this example, I'm assuming `cp` stands for "connector path" but, `cp` is also a common abbreviation for "copy" which was how my mind first read it.